README.md

```diff
@@ -1,87 +1,129 @@
+To send incidents, classify your Puppet server nodes with the `servicenow_reporting_integration::incident_mangement` class. The minimum parameters you need to configure for the class to work are instance (the fqdn of the servicenow instance, not including the protocol e.g. https://), and then user and password, or you can bypass username/password authentication and use an oauth token using the oauth_token parameter. Lastly you will need to get the sys_id of the user you would like to use as the ‘Caller’ for each ticket. Look below in the ‘How To’ section if you don’t already know how to get a user sys_id. The `servicenow_reporting_integration::incident_management` class requires the `caller_id` because that is a required incident field on ServiceNow’s end. 
 
-If you aren't ready to use Strings yet, manually create a REFERENCE.md in the root of your module directory and list out each of your module's classes, defined types, facts, functions, Puppet tasks, task plans, and resource types and providers, along with the parameters for each.
+To get the desired `sys_id` from Servicenow:
+1. In the Application Navigator (left sidebar navigation menu) navigate to System Security > Users and Groups > Users
+2. Use the search box to search for the user you want.
+3. In the user listing click on a user.
+4. In the user properties screen, click on the hamburger menu (three vertical bars) to the right of the left arrow in the upper left corner of the screen.
+5. Click on ‘Copy sys_id’ to copy the sys_id directly to the clipboard, or click on Show XML to see the sys_id in a new window along with the rest of the user’s properties.
 
-For each element (class, defined type, function, and so on), list:
+The `servicenow` report processor creates a ServiceNow incident based on that report if at least one of the incident creation conditions are met. 
 
-  * The data type, if applicable.
-  * A description of what the element does.
-  * Valid values, if the data type doesn't make it obvious.
-  * Default value, if any.
+By default the incident_creation_conditions are ['failures', 'corrective_changes']. This means an incident will be created if there was a resource failure, a catalog compilation failure, or if a corrective change was made to a managed resource during a Puppet run. With these default parameters intentional changes such as adding a new resource to the catalog and Puppet bringing it into compliance for the first time, and pending changes, like running an agent in noop mode and the agent reporting changes would have occurred, will not result in an incident in servicenow. If you would like to report on those types of changes please note that 'intentional_changes', 'pending_corrective_changes', and 'pending_intentional_changes' are also available as values for this parameter. As a shortcut you can also specify ‘always’, or to temporarily stop the module from creating any incidents at all you can set the incident_creation_conditions parameter to ‘never’.
```

Comment:
Would you mind wrapping the array ['failures', 'corrective_changes'] in back ticks to indicate that it's a piece of syntax that should be typed as shown?

Reply:
The additional values available for this param should probably also be wrapped in back ticks instead of single quotes.